executor/include/execution/trait/is_executor.hpp

```diff
@@ -0,0 +1,27 @@
+//
```

Comment:
Following PCL's syntax and using `impl/trait` might be better

Reply:
Not sure what will go in `impl` and what will go in the normal `.h` since everything is header only.

Reply:
The file which use includes is `.h`. Everything else is impl

Reply:
That seems like a good model. I still have some confusion. I will open up a separate issue for this discussion.